### DISCUSSION TITLE
Creating PET acquisition model without a template

### DISCUSSION THREAD

**Discussion by Kiarashdnsh** [SyneRBI/SIRF#1293] [Q&A]
Hi

I’m having trouble creating a PET acquisition model. I have read through most of the introductory files, and in all of them, a `template_sinogram.hs` is used to create the acquisition model. I don't have sinograms for my PET files, and using the provided template doesn't seem to be the correct approach for image reconstruction. How should I proceed from here?

I would greatly appreciate your help.

**Comment by KrisThielemans** [comment]
There's https://github.com/SyneRBI/SIRF/blob/master/examples/Python/PET/acquisition_data_from_scanner_info.py
but maybe you can elaborate what you want to do. If you have no sinograms, do you want to create some for a scanner that SIRF/STIR already supports? If so, the xample will help.

**Reply by Kiarashdnsh** [reply]
Thank you @KrisThielemans.

I looked up the link and managed to generate some files.

I want to generate sinograms for each PET image I possess to reconstruct attenuated PET images. However, the `*.hs` template file only generates from supported scanners, not the actual PET images. I also don't have the information on devices that acquired these images since the images I received were provided to me anonymously.

**Reply by KrisThielemans** [reply]
If you know nothing about the PET scanner, you'll have to make a reasonable choice. There's in STIR about axial spacing in PET images needing to correspond to the scanner, but that's common. In fact, the axial spacing will give some hints on which scanner they were acquired with. Feel free to post voxel-sizes and image shape here.

I'm still not so clear what you want to do. Just a caveat: if you have non-attenuation corrected (NAC) images and want to generate AC images, forward projection of the NAC images and reconstruction will not solve your problem (that kind of thing needs machine learning). But if you want to do simulations to assess impact etc, then this can work.

good luck

**Reply by Kiarashdnsh** [reply]
All my images are `256 * 256 * 256` with voxel sizes of `1 * 1 * 1`, with voxel values between 0 and 1 (I attached a sample interfile extracted by XMedCon). I want to create different varieties of low-res and high-res PET images from one original PET image for PVC, whether with AC or other methods possible with SIRF.

[m000-31013_AV451_S_E.zip](https://github.com/user-attachments/files/16887184/m000-31013_AV451_S_E.zip)

**Reply by KrisThielemans** [reply]
sorry, those dimensions are not your "standard" PET, so some resampling has gone. I cannot give you any further recommendations on what scanner to use. You'll have to pick one (and resample the images before projection). You'd need a mumap as well.

Best of luck

**Reply by Kiarashdnsh** [reply]
@KrisThielemans Yes, these images had undergone preprocessing. However, I have one original PET image. I can also request and gain access to others as well if I could make this work. This image has a timeframe of size `26`, dimensions of `256 * 256 * 127`, and voxel sizes of `1.4017 * 1.4017 * 2.0313`. I also attached the header of this file. (I can also get rid of the timeframe if it complicates things.)
Thanks!

 
[30001_ses-d2430_acq-AV45_pet.zip](https://github.com/user-attachments/files/16896698/30001_ses-d2430_acq-AV45_pet.zip)

**Reply by KrisThielemans** [reply]
that looks like a Siemens scanner. Have a look if you can find one in Scanner.cxx that has 64 rings (hence 127 image planes) and ring-distance 2*2.0313. There might be more than one of course.

best of luck

**Reply by Kiarashdnsh** [reply]
@KrisThielemans I looked up the `Scanner.cxx` master file. There's only one scanner with `64` rings and a ring spacing of `4.0625`, which is the Siemens mMR scanner. (If there are other files I must look up, please tell me.) I generated a template and ran the code. Got this error: 

```python
ERROR: ProjMatrixByBinUsingRayTracing sadly doesn't support shifted x/y origin yet  
.
.
.
error: ??? '"ProjMatrixByBinUsingRayTracing sadly doesn\'t support shifted x/y origin yet" exception caught at line 677 of /home/sirfuser/devel/buildVM/sources/SIRF/src/xSTIR/cSTIR/cstir.cpp; the reconstruction engine output may provide more information'
``` 

I used the original unprocessed PET image for this run.

Many thanks!

* Update: I get this error for almost all of the scanners I generate a template file with.
